dbclient/dbclient.py

```diff
@@ -312,9 +312,11 @@ def build_acl_args(self, full_acl_list, is_jobs=False):
                         current_owner = member.get('group_name')
 
         if is_jobs:
-            me = self.whoami()
-            if current_owner != me:
-                update_admin = {'user_name': self.whoami(),
+            # set current_owner to admin if current_owner is empty
+            # current_owner is empty when original job owner is obsolete user
+            current_owner = current_owner or admin_user_name
+            if current_owner != admin_user_name:
```

Comment:
@paulg-mscience 

I am confused by this change. The expected behavior is that we should append the `update_admin` block if this script runner (whoami) is not the current owner of the job. (whether it being empty or not)

In this changed behavior, if the current_owner is empty, then we are setting `current_owner = whoami` and it will make `current_owner == admin_user_name` in which case, it won't append the `update_admin` block which is not an expected behavior.

Could you explain what's going on?